addons/rose/addon/components/rose/code-editor/toolbar/index.js

```diff
@@ -0,0 +1,22 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { later } from '@ember/runloop';
+
+export const COPY_ICON_TYPE = 'flight-icons/svg/clipboard-copy-16';
+export const COPIED_ICON_TYPE = 'flight-icons/svg/clipboard-checked-16';
+
+export default class RoseCodeEditorToolbarComponent extends Component {
+  @tracked copyIconType = COPY_ICON_TYPE;
+
+  @action
+  copied() {
+    let { onCopy } = this.args;
+    let originalIconType = this.copyIconType;
+
+    this.copyIconType = COPIED_ICON_TYPE;
+    onCopy && onCopy();
```

Comment:
Would an `if` statement be more consistent and/or clearer?

Reply:
I'll update this to use an `if` statement if that's our convention but I personally think it's pretty clear, I would say it's quite idiomatic JS

Reply:
It is a conditional yes, but I don't think we use this convention elsewhere at this time.